airflow/plugins_manager.py

```diff
@@ -75,15 +78,16 @@ def validate(cls):
             # normalize root path as namespace
             namespace = '_'.join([re.sub(norm_pattern, '__', root), mod_name])
 
-            m = imp.load_source(namespace, filepath)
-            for obj in list(m.__dict__.values()):
-                if (
-                        inspect.isclass(obj) and
-                        issubclass(obj, AirflowPlugin) and
-                        obj is not AirflowPlugin):
-                    obj.validate()
-                    if obj not in plugins:
-                        plugins.append(obj)
+            with timeout(plugin_import_timeout):
```

Comment:
I think that you will need to catch the `AirflowTaskTimeout` that is raised during a timeout, log something, and continue.